scripts/mapcheck.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+
+# Check if hostname mappings in provided map.yml matches the domain's
+# `_onion-mx._tcp.domain.tld.` SRV records. (see SRV.md for details)
+#
+# If all queries fail, your DNS resolver may not work as expected.
+#
+# requirements: dnspython pyyaml
+
+import sys
+import yaml, dns.resolver
+
+
+if len(sys.argv) != 2:
+    print("usage: {} path/to/map.yml".format(sys.argv[0]))
+    exit(1)
+
+mapfilename = sys.argv[1]
+resolver = dns.resolver.Resolver()
+fails = 0
+
+with open(mapfilename, 'r') as mapfile:
+    mapdata = yaml.load(mapfile)
+total = sum((len(v) for k,v in mapdata.items()))
+
+for onion,domains in mapdata.items():
+    for domain in domains:
+        name = "_onion-mx._tcp.{}".format(domain)
+        try:
+            response = resolver.query(name, "SRV")
+            result = [str(x).rstrip(".").split(" ")[-1] for x in response]
+            if not onion in result:
+                print("FAIL {} -> {}".format(domain, onion))
+                fails += 1
+        except Exception as e:
+            print("FAIL {} -> {} ({})".format(domain, onion, e))
+            fails += 1
+
+# report back to the user
+print("INFO {} of {} queries failed".format(fails, total), file=sys.stderr)
```

Comment:
Instead of printing for each failure, you can use a list of failed and for each failure, append in the list the domain+onion (something like `fails.append("{} != {}".format(domain, onion))`. Then here in the end, 
```
if fails:
    print("INFO {} of {} queries failed".format(len(fails), total), file=sys.stderr)
    print("\n".join(b))
    exit(2)

exit(0)
```
So if everything is fine, no prints and exit(0), else print a statistic (x of y queries failed) and which failures did we have

Reply:
I don't see an advantage in putting them in a list first. But I will use exit(2) if there were any failures.